pandas/core/internals/blocks.py

```diff
@@ -1809,6 +1809,11 @@ def _slice(self, slicer):
             if not com.is_null_slice(slicer[0]):
                 raise AssertionError("invalid slicing for a 1-ndim categorical")
             slicer = slicer[1]
+        elif not isinstance(slicer, tuple) and self.ndim == 2:
+            # reached via getitem_block via _slice_take_blocks_ax0
+            # TODO(EA2D): wont be necessary with 2D EAs
+            # treat this like (slicer, slice(None)
```

Comment:
I find this hard to grasp without knowing what exactly what those other methods do. Could you update the `_slice` docstring to indicate what kind of "slicer" objects get passed here?

Reply:
will take a look

Reply:
Looks like either slice object or tuple([None, slice_object]) or mask.  not very helpful since thats basically "most of what you can slice things with"

Reply:
At least putting something like that in the docstring (even vague) is useful I think.